src/tooling/grain_doc.md

```diff
@@ -0,0 +1,108 @@
+---
+title: grain doc
+---
+
+Grain ships with a documentation generator built directly into the compiler!
+
+You may be used to similar tools, such as `JSDoc`, that allow you to add a comment above code which describes the input and output types, along with other documentation. Generally, you'll run an external tool against your source code to generate your documentation.
+
+In Grain, we have provided this as another command via the CLI, and it inherits the full power of the compiler. For example, you won't need to specify any types in your doc comments, as the compiler will infer them. 🎉
+
+Here's a quick example of adding a doc block to your grain code:
+```gr
+/**
+ * An alias for normal syntactic array access, i.e. `array[n]`.
+ *
+ * Retrieves the element from the array at the specified index.
+ * A negative index is treated as an offset from the end of the array.
+ *
+ * @param index: The index to access
+ * @param array: The array to access
+ * @returns The element from the array
+ *
+ * @since v0.1.0
+ * @history v0.2.0: Argument order changed to data-last
+ */
+export let get = (index, array) => {
+  array[index]
+}
+```
+
+## Usage
+
+Before running `grain doc`, you'll need to add the doc block comments to your code. Doc block comments are started with `/**` and ended with `*/`. They must exist on the line above your type declaration or exported values. No docs will be generated for non-exported values so you can keep internals private.
```

Comment:
```suggestion
Before running `grain doc`, you'll want to add the doc block comments to your code. Doc block comments are started with `/**` and ended with `*/`. They must exist on the line above your type declaration or exported values. No docs will be generated for non-exported values so you can keep internals private.
```

"want" instead of "need" because you can run the command without any comments